kuzzleio/kuzzle-admin-console
# KZL-694 Refresh search when hitting "Search"

## What does this PR do ?
fixes #467 
Before, the DocumentList page watched the `currentFilter` object for changes to trigger the `fetchDocuments` function. I changed this behavior, now the `fetchDocuments` function is triggered every time the "search" button is clicked.

### How should this be manually tested?
1. Launch local End to End tests with `npm run e2e-local`
2. Start the `search.spec.js` spec file
3. See what happens in the `refreshes search when the Search button is hit twice` test case
4. Reproduce it in your favorite browser.

### Other changes
The test framework was a little bit broken. Now it's meant to be more stable.
